x-pack/legacy/plugins/ml/index.ts

```diff
@@ -60,43 +50,8 @@ export const ml = (kibana: any) => {
       },
     },
 
-    async init(server: MlServer) {
-      const kbnServer = (server as unknown) as KbnServer;
-
-      const initializerContext = ({
-        legacyConfig: server.config(),
-        logger: {
-          get(...contextParts: string[]) {
-            return kbnServer.newPlatform.coreContext.logger.get('plugins', 'ml', ...contextParts);
-          },
-        },
-      } as unknown) as MlInitializerContext;
-
-      const mlHttpService: MlHttpServiceSetup = {
-        ...kbnServer.newPlatform.setup.core.http,
-        route: server.route.bind(server),
-      };
-
-      const core: MlCoreSetup = {
-        injectUiAppVars: server.injectUiAppVars,
-        http: mlHttpService,
-        savedObjects: server.savedObjects,
-        coreSavedObjects: kbnServer.newPlatform.start.core.savedObjects,
-        elasticsearch: kbnServer.newPlatform.setup.core.elasticsearch,
-      };
-      const { usageCollection, cloud, home } = kbnServer.newPlatform.setup.plugins;
-      const plugins = {
-        elasticsearch: server.plugins.elasticsearch, // legacy
-        security: server.newPlatform.setup.plugins.security,
-        xpackMain: server.plugins.xpack_main,
-        spaces: server.plugins.spaces,
-        home,
-        usageCollection: usageCollection as UsageCollectionSetup,
-        cloud: cloud as CloudSetup,
-        ml: this,
-      };
-
-      plugin(initializerContext).setup(core, plugins);
+    async init(server: Server) {
```

Comment:
`managementSections: ['plugins/ml/application/management'],` on line 24 isn't registering our management section anymore.
Is there a NP equivalent? 
(github wouldn't let me add this to line 24)

Reply:
There is, but I think it should work. Maybe this this an issue with the fact that there is now both a `x-pack/legacy/plugins/ml` and `x-pack/plugins/ml` folders? But the NP `ml` plugin has no `public` part, so I'm not sure to understand. @elastic/kibana-app-arch would you mind to take a look?

Reply:
@jgowdyelastic - looks like the issue is that in `applicatoins/management` the code to register the app in management is wrapped in an if block that relies on legacy `xpackInfo`. 

The solution is to use the frontend licensing service -  `npSetup.plugins.licensing` where `npSetup`  comes from `import { npSetup } from 'ui/new_platform'`

Then we can can pull the relevant license info from 
```
licensing.license$.subscribe(async license => {
  if (license.type && VALID_FULL_LICENSE_MODES.includes(license.type)) {
    ... 
```

This can be added in a follow-up - not a big change